src/domain_classes/tree_node.py

```diff
@@ -266,12 +266,21 @@ def __init__(
         )
         self.entity: dict = entity if entity else {}
         self.error_message = None
+        if (
+            entity is not None
+            and entity != {}
+            and entity["type"] == "dmss://system/SIMOS/Link"
+            and "targetName" in entity
+        ):
+            self.type = entity["targetType"]
 
     def is_root(self):
         return super().is_root()
 
     @property
     def name(self):
+        if "targetName" in self.entity and self.entity["type"] == "dmss://system/SIMOS/Link":
+            return self.entity.get("targetName", self.attribute.name)
```

Comment:
Is this really needed?
Where are we using the "name" property? In theory there should be no hard coded dependencies to it.
I vote for removing the whole property function

Reply:
I think removing it will require a bit of work.

the name property is used several places in document_service.

But I can add a new issue for it!